### PR TITLE
feat(profiling): dont disable empty thread profiles

### PR DESCRIPTION
**Pull request by eliashussary**
Closes: https://github.com/getsentry/team-profiling/issues/59

Empty profiles in the thread selector were previously disabled. We now allow them to be selected and default to the empty state view within the flamechart.

image

image

### CODE REVIEW

**Review by JonasBa — APPROVED** [review]
Should we add some divider in the dropdown that says "Profiles with 0 samples" or something so they are "stowed" away?

**Comment by eliashussary** [comment]
> Should we add some divider in the dropdown that says "Profiles with 0 samples" or something so they are "stowed" away?

yeah that's a great idea

File: static/app/components/profiling/threadSelector.tsx [hunk 1]
```diff
@@ -58,7 +72,21 @@ function ThreadMenuSelector({
...
-      options={options}
+      options={[
+        {
+          // TODO: fix CompactSelect types to better represent usage.
+          // this value has no effect on the selection, it's simply to satisfy the types..
+          // tried modifying the types but it creates a lot of errors
+          value: threadId || 0,
+          label: t('Profiles'),
+          options: profileOptions,
+        },
```
Comment (eliashussary) [inline static/app/components/profiling/threadSelector.tsx:83]: tsc errors when modifying `options` to support `Array<{label:string; options: OptionType[]}>`
https://github.com/getsentry/sentry/actions/runs/3491715766/jobs/5844677026

File: static/app/components/profiling/threadSelector.tsx [hunk 2]
```diff
@@ -58,7 +72,21 @@ function ThreadMenuSelector({
         icon: <IconList size="xs" />,
         size: 'xs',
       }}
-      options={options}
+      options={[
+        {
+          // TODO: fix CompactSelect types to better represent usage.
+          // this value has no effect on the selection, it's simply to satisfy the types..
+          // tried modifying the types but it creates a lot of errors
+          value: threadId || 0,
```
Comment (Zylphrex) [inline static/app/components/profiling/threadSelector.tsx:80]: I don't remember but I think you want to use a different `value` for the 2 nested options or the compact select might get confused..

**Pull request merged**